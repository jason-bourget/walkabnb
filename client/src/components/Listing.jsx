import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import red from '@material-ui/core/colors/red';
import FavoriteIcon from '@material-ui/icons/Favorite';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';

const styles = theme => ({
  card: {
    maxWidth: 550,
    margin: 10
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
  actions: {
    display: 'flex',
  },
  avatar: {
    backgroundColor: red[500],
  },
});

class Listing extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      expanded: false
    }
  }

  handleExpandClick = () => {
    this.setState(state => ({ expanded: !state.expanded }));
  }

  render() {
    const { classes } = this.props;
    const { title, url, walkscore, size, reviews, rating, image, city} = this.props.listing;
    console.log(title);
    return (
      <Card className={classes.card}>
        <CardHeader
          avatar={
            <Avatar aria-label="Walkscore" className={classes.avatar}>
              {walkscore}
            </Avatar>
          }
          title={title}
          subheader={city}
        />
        <CardMedia
          className={classes.media}
          image={image}
          title="Listing Image"
        />
        <CardContent>
          <Grid container className={classes.demo} justify="space-evenly">
            {size.map(value => (
              <Grid key={value} item>
                <Typography component="p">
                  {value}
                </Typography>
              </Grid>
            ))}
          </Grid>
          <Grid container className={classes.demo} justify="space-evenly">
            <Grid item>
              <Typography component="p">
                {rating} out of 5 stars based on {reviews} reviews
              </Typography>
            </Grid>
            <Grid item>
              <Typography component="p">
                $99/night
              </Typography>
            </Grid>
          </Grid>
        </CardContent>
        <CardActions className={classes.actions} disableActionSpacing>
          <IconButton aria-label="Add to favorites">
            <FavoriteIcon />
          </IconButton>
        </CardActions>
      </Card>
    )
  }
}

export default withStyles(styles)(Listing);