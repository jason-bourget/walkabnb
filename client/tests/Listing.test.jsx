import React from 'react';
import Listing from '../src/components/Listing.jsx';
import Card from '@material-ui/core/Card';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

const listing = {
  airbnbId: 13815789,
  city: "denver",
  coordinates: ["39.73175", "-104.94928"],
  createdAt: "2019-05-11T03:34:58.241Z",
  id: 54,
  image: "https://a0.muscache.com/im/pictures/7ca1d55d-9f8a-47f1-9819-b4970df3825e.jpg",
  price: 59,
  rating: "5",
  reviews: 272,
  size: ["6 guests", "3 bedrooms", "5 beds", "1 bath"],
  title: "Cozy Craftsman Bungalow near City Center",
  updatedAt: "2019-05-11T03:34:58.241Z",
  url: "https://www.airbnb.com/rooms/13815789",
  walkscore: 70
}

test('renders correctly', () => {
  const component = renderer.create(
    <Listing listing={listing}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Listing component', () => {

  it('should render the correct price', () => {
    const wrapper = mount(<Listing listing={listing}/>);
    expect(wrapper.containsMatchingElement(
      <div>$59/night</div>
    )).toBe(true);
  });

  it('should render the correct description', () => {
    const wrapper = mount(<Listing listing={listing}/>);
    expect(wrapper.containsMatchingElement(
      <span className="MuiTypography-root-41 MuiTypography-body2-49 MuiCardHeader-title-36">
        Cozy Craftsman Bungalow near City Center
      </span>
    )).toBe(true);
  });
});
