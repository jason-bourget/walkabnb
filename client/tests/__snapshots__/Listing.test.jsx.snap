// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation1-8 MuiPaper-rounded-6 MuiCard-root-4 Listing-card-1"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="MuiCardHeader-root-32"
  >
    <div
      className="MuiCardHeader-avatar-33"
    >
      <div
        aria-label="Walkscore"
        className="MuiAvatar-root-38 MuiAvatar-colorDefault-39"
        style={
          Object {
            "backgroundColor": "#2196f3",
          }
        }
      >
        70
      </div>
    </div>
    <div
      className="MuiCardHeader-content-35"
    >
      <span
        className="MuiTypography-root-41 MuiTypography-body2-49 MuiCardHeader-title-36"
      >
        Cozy Craftsman Bungalow near City Center
      </span>
      <span
        className="MuiTypography-root-41 MuiTypography-body2-49 MuiTypography-colorTextSecondary-74 MuiCardHeader-subheader-37"
      >
        denver
      </span>
    </div>
  </div>
  <a
    href="https://www.airbnb.com/rooms/13815789"
    target="_blank"
  >
    <div
      className="MuiCardMedia-root-77 Listing-media-2"
      style={
        Object {
          "backgroundImage": "url(\\"https://a0.muscache.com/im/pictures/7ca1d55d-9f8a-47f1-9819-b4970df3825e.jpg\\")",
        }
      }
      title="Listing Image"
    />
  </a>
  <div
    className="MuiCardContent-root-79"
    style={
      Object {
        "backgroundColor": "#2196f3",
      }
    }
  >
    <div
      className="MuiGrid-container-80 MuiGrid-justify-xs-space-evenly-101"
    >
      <div
        className="MuiGrid-item-81"
      >
        <div
          className="MuiTypography-root-41 MuiTypography-subtitle2-60"
          style={
            Object {
              "color": "white",
            }
          }
        >
          6 guests
        </div>
      </div>
      <div
        className="MuiGrid-item-81"
      >
        <div
          className="MuiTypography-root-41 MuiTypography-subtitle2-60"
          style={
            Object {
              "color": "white",
            }
          }
        >
          3 bedrooms
        </div>
      </div>
      <div
        className="MuiGrid-item-81"
      >
        <div
          className="MuiTypography-root-41 MuiTypography-subtitle2-60"
          style={
            Object {
              "color": "white",
            }
          }
        >
          5 beds
        </div>
      </div>
      <div
        className="MuiGrid-item-81"
      >
        <div
          className="MuiTypography-root-41 MuiTypography-subtitle2-60"
          style={
            Object {
              "color": "white",
            }
          }
        >
          1 bath
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiCardContent-root-79"
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "paddingBottom": "16px",
      }
    }
  >
    <div
      className="MuiGrid-container-80 MuiGrid-justify-xs-space-evenly-101"
    >
      <div
        className="MuiGrid-item-81"
      >
        <div
          className="MuiTypography-root-41 MuiTypography-subtitle2-60"
        >
          $
          59
          /night
        </div>
      </div>
      <div
        className="MuiGrid-item-81"
      >
        <div
          className="MuiTypography-root-41 MuiTypography-subtitle2-60"
        >
          Rating: 
          5
          /5
        </div>
      </div>
      <div
        className="MuiGrid-item-81"
      >
        <div
          className="MuiTypography-root-41 MuiTypography-subtitle2-60"
        >
          272
           reviews
        </div>
      </div>
    </div>
  </div>
</div>
`;
